### DISCUSSION TITLE
Issue/Challenge with conditional rendering

### DISCUSSION THREAD

**Discussion by Mondal10** [vueform/vueform#132] [Questions] [answered]
I am facing an issue/challenge related to nesting of conditions in conditional rendering properties for the form fields.

eg: Condition that is to be set: name = "Test" OR (age between 19 AND 25)
```
[
    [
      ['name', '==', 'Test'],
      [['age', '>', 19], ['age', '<', 25]], // age between (19 and 26), since there is no built-in between operator
    ],
]
```

In this case the second scenario does not get triggered as the conditions property does not support the nesting.

## Alternatives that are tried:
- Tried [Functional Conditions](https://vueform.com/docs/conditional-rendering#functional-conditions), but it also has some limitations,
eg:
```
// Working fine
// Condition: name = "Test" AND between age custom function
[
    [
        'name',
        "==",
        "Test"
    ],
    function between(form$)
]

// Does not work (since there is nesting of array it does not check for function)
// Condition: name = "Test" OR between age custom function 
[
    [
        [
            'name',
            "==",
            "Test"
        ],
        function between
    ]
]
```

- Tried Using just a single JSON rule engine function that will check and return condition result but [JSON rule engine response is a promise](https://github.com/CacheControl/json-rules-engine/blob/master/docs/engine.md#enginerunobject-facts-object-options---promise--events--failureevents--almanac-almanac-results--failureresults-) and condition props in Vueforms does not support async function and considers the returned promise as a truthy value so the fields are always visible irrespective of result.

I have thoroughly reviewed the documentation but couldn't find a clear resolution to this problem. If there are any best practices or workarounds that I might have missed, please guide me in the right direction.

**Comment by adamberecz** [comment, marked as answer]
In this case you can use only a `function` as a condition:

```js
[
    (form$) => {
      return form$.el$('name')?.value === 'Test' || (form$.el$('age')?.value > 19 && form$.el$('age')?.value < 25)
    }
]
```

Nevertheless I added `between` rule in `1.7.2` and `AND` groups in `OR` groups, so your first example should work as well.

**Reply by Mondal10** [reply]
Hi, thanks for the solution. Actually this was just an simple example, I have couple of more custom functions that needs to be taken care of using JSON rule engine.

Can we have a support for async function in the condition ?

**Reply by adamberecz** [reply]
Unfortunately we can't have async functions in conditions. The `available` property is a computed that comes from checking conditions real time (on each data model change). Imagine having that with async calls.
